### PR TITLE
rename 'redis' to 'server'

### PR DESCRIPTION
**Pull request by 9bany**
https://github.com/valkey-io/valkey/issues/144#issuecomment-2035356220

### CODE REVIEW

File: src/cluster_legacy.c [hunk 1]
```diff
@@ -5242,12 +5242,12 @@ void clusterSetMaster(clusterNode *n) {
  * Nodes to string representation functions.
  * -------------------------------------------------------------------------- */
 
-struct redisNodeFlags {
+struct serverNodeFlags {
```
Comment (madolson) [inline src/cluster_legacy.c]:
```suggestion
struct clusterNodeFlags {
```
Server doesn't really make sense here, since they're cluster node flags.

**Review by madolson — CHANGES_REQUESTED** [review]

File: src/cluster_legacy.c [hunk 2]
```diff
@@ -5242,12 +5242,12 @@ void clusterSetMaster(clusterNode *n) {
...
  * -------------------------------------------------------------------------- */
 
-struct redisNodeFlags {
+struct clusterNodeFlags {
     uint16_t flag;
     char *name;
 };
 
-static struct redisNodeFlags redisNodeFlagsTable[] = {
+static struct serverNodeFlags serverNodeFlagsTable[] = {
```
Comment (madolson) [inline src/cluster_legacy.c]:
```suggestion
static struct clusterNodeFlags clusterNodeFlagsTable[] = {
```

File: src/cluster_legacy.c [hunk 3]
```diff
@@ -5262,9 +5262,9 @@ static struct redisNodeFlags redisNodeFlagsTable[] = {
  * string 'ci'. */
 sds representClusterNodeFlags(sds ci, uint16_t flags) {
     size_t orig_len = sdslen(ci);
-    int i, size = sizeof(redisNodeFlagsTable)/sizeof(struct redisNodeFlags);
+    int i, size = sizeof(serverNodeFlagsTable)/sizeof(struct serverNodeFlags);
```
Comment (madolson) [inline src/cluster_legacy.c]:
```suggestion
    int i, size = sizeof(clusterNodeFlagsTable)/sizeof(struct clusterNodeFlags);
```

File: src/cluster_legacy.c [hunk 4]
```diff
@@ -5262,9 +5262,9 @@ static struct redisNodeFlags redisNodeFlagsTable[] = {
...
     for (i = 0; i < size; i++) {
-        struct redisNodeFlags *nodeflag = redisNodeFlagsTable + i;
+        struct serverNodeFlags *nodeflag = serverNodeFlagsTable + i;
```
Comment (madolson) [inline src/cluster_legacy.c]:
```suggestion
        struct clusterNodeFlags *nodeflag = clusterNodeFlagsTable + i;
```

**Review by madolson — COMMENTED** [review]
Sorry, I should have mentioned to update all occurrences of it. The build will fail.

**Comment by 9bany** [comment]
@0del Well done

**Review by madolson — APPROVED** [review]

**Pull request merged**